bin/setup

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+##
+# Setup script to symlink configuration and dot files from this repository
+# to proper places in the system.
+#
+# Script is idempotent. It is same to run it several times.
+#
+# TODO:
+
+# Stop immediately if any external program returns a non-zero exit status.
+# Short: set -e
+set -o errexit
+# Treat unset variables as an error when substituting. Short: set -u
+set -o nounset
+# If set, the return value of a pipeline is the value of the last (rightmost)
+# command to exit with a non-zero status, or zero if all commands in the
+# pipeline exit successfully. This option is disabled by default.
+set -o pipefail
+
+# Uncomment for debugging
+#
+# Disable file name generation using metacharacters (globbing).
+# Short: set -f
+#set -o noglob
+# Prints shell input lines as they are read.
+# Short: set -v
+#set -o verbose
+# Print command traces before executing command.
+# Short: set -x
+#set -o xtrace
+
+
+readonly __DIR__="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly __LIB__="${__DIR__}/../lib-bash"
+
+readonly __CONFIG__="${__DIR__}/../CONFIG"
+
+
+readonly USER_HOME_DIR_PLACEHOLDER="USER_PLACEHOLDER"
+
+readonly DEFAULT_USER="$USER"
+declare  USER_NAME="${DEFAULT_USER}"
+
+readonly DEFAULT_HOME_DIR_PATH="$HOME"
+declare  HOME_DIR_PATH="${DEFAULT_HOME_DIR_PATH}"
+
+readonly DEFAULT_CONFIG_DIR_NAME=".config"
+declare  CONFIG_DIR_NAME="${DEFAULT_CONFIG_DIR_NAME}"
+
+
+main() {
+  while getopts "H:U:h" _FLAG; do
+    case $_FLAG in
+      U)
+        USER_NAME="$OPTARG"
+        ;;
+      H)
+        HOME_DIR_PATH="$OPTARG"
```

Comment:
Same regarding double quotes:) (and 3 lines above)

Reply:
Addressed by 4532f2df22a5565f858d084737526a814d8c419f.